src/main/java/com/alkemy/ong/repository/UserRepository.java

```diff
@@ -10,4 +10,9 @@
 public interface UserRepository extends CrudRepository<UserModel,Long> {
 
     List<UserModel> findAll();
+
+    UserModel findByEmail (String email);
+
+    boolean existsByEmail (String email);
```

Comment:
Estos dos métodos los podes combinar en uno solo que sea `findByEmail` y que retorne un `Optional<UserModel>`. De ese modo, cuando lo busques, luego chequeas si `isPresent()`: si no lo está, tiras la excepcion.

Reply:
Santi  a mi me parece mas claro la forma en la que lo escribí yo, tal vez vos con tu experiencia crees que no, pero hasta esta bueno que uno te devuelve un boolean y otro un UserModels, yo no suelo usar el isPresent(), pero si me decis que lo cambie, lo cambio

Reply:
No hay problema 👌🏼  en estos casos no va a pasar nada, porque vamos a tener pocos registros, pero en una aplicación que tiene muchos registros y mucho tráfico, al hacer primero una busqueda para ver si existe, y luego traerte el registro, estas haciendo dos consultas a la base de datos, y eso puede ser muy costoso.
Teniendo solo un `Optional<UserModel> findByEmail(String email);` podes hacer:
```suggestion
    public void loginUser(User user) {
    Optional<UserModel> optionalUser = userRepository.findByEmail(user.getEmail());
    if (!optionalUser.isPresent()) {
         throw UserNotFoundException();
    }
    return getUserLoginChecked(user, optionalUser);
```